### PR TITLE
A slightly weird work around to prevent importing sunpy.database in sunpy.map

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: sunpy/util/types.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+"""
+A collection of ABCs to use to shortcut having to import things.
+"""
+import abc
+
+class DatabaseEntryType(metaclass=abc.ABCMeta):
+    """
+    A Base class for sunpy.database.DatabaseEntry.
+    """
```
Comment (nabobalis) [inline sunpy/util/types.py]: You envision more that might need to be in utils? I feel like this could have been in database folder but swings and cars.

Comment (Cadair) [reply sunpy/util/types.py]: If you put it in the database folder then you have to execute `database/__init__.py` which costs most of the import time.

Comment (nabobalis) [reply sunpy/util/types.py]: Good point, I did not think of that.

**Comment by dstansby** [comment]
This is very cunning 🦊

**Pull request merged**